src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -199,366 +317,528 @@ static AnimationInfoS sAnimationInfo[] = {
+void EnGo_GraveBroGoron_UpdateToSurprisedAnimation(EnGo* this, PlayState* play);
+
+s32 EnGo_UpdateRotationToTarget(EnGo* this, PlayState* play);
+
+s32 EnGo_UpdateSfx(EnGo* this, PlayState* play);
+
+s32 EnGo_UpdateFocus(EnGo* this);
+
+void EnGo_UpdateCollider(EnGo* this, PlayState* play);
+void EnGo_UpdateSnowballCollider(EnGo* this, PlayState* play);
+void EnGo_UpdateRolledUpCollider(EnGo* this, PlayState* play);
+void EnGo_UpdateFrozenCollider(EnGo* this, PlayState* play);
+void EnGo_UpdateStandingCollider(EnGo* this, PlayState* play);
+void EnGo_PowderKegGoron_UpdateCollider(EnGo* this, PlayState* play);
+s32 EnGo_CanSnowballHurtPlayer(PlayState* play);
+
+void EnGo_UpdateEffects(EnGo* this);
+void EnGo_UpdateSnow(EnGoEffect* pEffect, f32 dustConversionHeight);
+void EnGo_MakeSteam(EnGo* this);
+void EnGo_InitSnow(EnGoEffect* pEffect, Vec3f worldPos);
+void EnGo_InitDust(EnGoEffect* pEffect, Vec3f pos, Vec3f accel, Vec3f vel, f32 scale, f32 deltaScale, s32 maxFrames,
+                   EnGoEffectType parentEffectType);
+EnGoEffect* EnGo_InitSteam(EnGoEffect* pEffect, Vec3f position, Vec3f acceleration, Vec3f velocity, f32 scale,
+                           f32 deltaScale, s32 maxFrames);
+
+void EnGo_CheckAndDrawIceBlock(EnGo* this, PlayState* play, f32 scale, f32 alpha);
+void EnGo_DrawSnow(EnGoEffect* pEffect, PlayState* play, Gfx* material, Gfx* model, u8 effectType);
+void EnGo_DrawDust(EnGoEffect* pEffect, PlayState* play2);
+void EnGo_DrawSteam(EnGoEffect* pEffect, PlayState* play2);
+void EnGo_DrawEffects(EnGo* this, PlayState* play);
+
+/**
+ * Initialize the 2nd Half of the effect objects to be of type ENGO_EFFECT_STEAM.
+ *
+ * @param pEffect Pointer to the latter half of the EnGoEffect Table (effectTable[ENGO_NUM_SNOW_EFFECTS]).
+ * @param position Initial position
+ * @param acceleration Initial acceleration
+ * @param velocity Initial velocity
+ * @param scale Initial scale
+ * @param deltaScale How much the effect scales per frame (additive)
+ * @param maxFrames Maximum number of frames the effect will last. Actual lifetime will be 1/3 * maxFrames -> maxFrames
+ */
+EnGoEffect* EnGo_InitSteam(EnGoEffect* pEffect, Vec3f position, Vec3f acceleration, Vec3f velocity, f32 scale,
+                           f32 deltaScale, s32 maxFrames) {
     s32 i;
 
-    for (i = 16; i < 32; i++, ptr++) {
-        if (ptr->unk_00 == 0) {
-            ptr->unk_00 = 7;
-            ptr->unk_01 = (Rand_ZeroOne() * (2.0f * (arg6 / 3.0f))) + (arg6 / 3.0f);
-            ptr->unk_02 = ptr->unk_01;
-            ptr->unk_10 = arg1;
-            ptr->unk_1C = arg2;
-            ptr->unk_28 = arg3;
-            ptr->unk_34 = arg4;
-            ptr->unk_38 = arg5;
-            break;
-        }
-    }
+    // Steam effects are from the end of the snow effects to the end of all effects
+    for (i = ENGO_NUM_SNOW_EFFECTS; i < ENGO_NUM_EFFECTS; i++, pEffect++) {
+        if (pEffect->effectType != ENGO_EFFECT_NONE)
+            continue;
```

Comment:
```suggestion
        if (pEffect->effectType != ENGO_EFFECT_NONE) {
            continue;
        }
```
clang-tidy should be catching this, might want to double check you have it installed.
There are also more of these that need to be fixed below

Reply:
Installed clang-tidy and it then did correct this with ./format.sh Thanks for helping catch that!